Automattic/woocommerce-payments
# Disable payment gateway when not in test mode and not using https or ssl checkout enforcement

Fixes #2374 

#### Changes proposed in this Pull Request
PHP side:
- `is_available` method now checks for not being in test mode, using `https` or having the ssl in checkout enforced.
-  Passes `woocommerce_force_ssl_checkout` to React in order to use it in the Task list.

JS side:
- Adds a new Task if not in Test mode, not using https or not having woocommerce_force_ssl_checkout enabled.
-  The new Task uses `actionLabel` and `action`. The old tasks will need to fix `action` usage (see #2479).

![image](https://user-images.githubusercontent.com/2612426/126162334-b3e36b3d-0a57-4d9f-83b3-f2d3764f7049.png)

#### Testing instructions





* Disable Test mode
* Disable SSL checkout enforcement here http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=advanced
* Go to Payments Overview and see the task active
* Try to buy something
* Go to checkout
* See payment gateway disabled in test mode and without https or ssl checkout enforcement:  `Sorry, it seems that there are no available payment methods for your state. Please contact us if you require assistance or wish to make alternate arrangements.`
* Enable SSL checkout enforcement
* Go back to Payments, see task not showing up


-------------------
